include/core/memory.hpp

```diff
@@ -247,12 +282,52 @@ class Memory {
   std::map<std::size_t, std::size_t> _protection{};
 
   /**
+   * \brief Returns a MemoryValue holding the data stored in the Memory at
+   *        [address; address+amount[, undefined behaviour if
+   *        [address; address+amount[ is not subset of [0;_byteCount[
+   * \param address Starting location of the Value in the Memory
+   * \param amount Number of Bytes comprising the value
+   * \returns MemoryValue holding the data stored in the Memory at
+   *          [address;address+amount[
+   */
+  MemoryValue _get(std::size_t address, std::size_t amount = 1) const;
+  /**
+   * \brief Writes value into the Memory at address, undefined behaviour if
+   *        [address; address + value.size() / _byteSize[ is not subset of
+   *        [0;_byteCount[
+   * \param address Starting address of the to be overwritten value
+   * \param value Value to write
+   * \param amount precalculated amount of bytes stored in value
+   * \param ignoreProtection if this is true do ignore all protection
+   */
+  void _put(std::size_t address,
+            const MemoryValue &value,
+            std::size_t amount,
+            bool ignoreProtection = false);
+  /**
+   * \brief Writes value into the Memory at address and returns the previous
+   *        value, undefined behaviour if [address; address + value.size() /
+   *        _byteSize[ is not subset of [0;_byteCount[
+   * \param address Starting address of the to be overwritten value
+   * \param value Value to write
+   * \param ignoreProtection if this is true do ignore all protection
+   * \returns Value that was overwritten
+   */
+  MemoryValue _set(std::size_t address,
+                   const MemoryValue &value,
+                   bool ignoreProtection = false);
+  /**
+   * \brief This Method is called when the whole Memory changes and
+   *        notifies the Gui of the change
+   */
+  void _wasUpdated();
+  /**
    * \brief This Method is called whenever something in the Memory changes and
    *        notifies the Gui of the change
    * \param address address of the updated value
    * \param amount length of the updated value
    */
-  void _wasUpdated(const std::size_t address, const std::size_t amount = 1);
```

Comment:
The ordering of things in this file is all off. It should be:

1. Typedefs and nested type declarations
2. Static variables
3. Static functions
4. Private member functions
5. Private variables